terratorch/tasks/base_task.py

```diff
@@ -50,6 +54,64 @@ def configure_models(self) -> None:
         if self.hparams["freeze_head"]:
             self.model.freeze_head()
 
+    def handle_full_or_tiled_inference(self, x, num_categories:int=None, **rest):
+
+        # When the input sample cannot be fit on memory for some reason
+        # the tiled inference is automatically invoked.
+        def model_forward(x,  **kwargs):
+            return self(x, **kwargs).output
+
+        # The kind of memory device we are considering 
+        if torch.cuda.is_available():
+            device = "GPU Memory"
+        else:
+            device = "RAM"
+
+        if not self.tiled_inference_on_testing:
+            # When the user don't set the variable `tiled_inference_on_testing`
+            # as `True` in the config, we will try to use full inference.
+            try:
+                model_output: ModelOutput = self(x, **rest)
+            except (torch.OutOfMemoryError, MemoryError)  as e:
+                raise Exception("Inference on testing failed due to memory issues. Try to pass `tiled_inference_on_testing` as `True`, to use tiled inference for it.")
+
+        else:
+            try:
+                # Even when `tiled_inference_on_testing` is `True`, we at least
```

Comment:
I would not do that. This is a bit of black magic. The framework is doing things which the user not explicitly requested to do. So users could get confused if they think they run tiled inference on one system but in the background it's not and are happy with the model performance, then on a different system it will run into the OOM error and deliver inferior results 

@paolo-fraccaro @Joao-L-S-Almeida agree?